src/ductile/schemas.clj

```diff
@@ -5,21 +5,28 @@
   (:import [org.apache.http.impl.conn PoolingClientConnectionManager
             PoolingHttpClientConnectionManager]))
 
+(s/defschema RequestFn
+  "A function implementing the 1-argument
+  arity of clj-http.client/request."
+  (s/=> s/Any (s/named s/Any 'req)))
```

Comment:
we could perhaps have an open schema with the minimal set of keys that we do use
```clojure
(st/open-schema
  {:url s/Str
   :method (s/enum :get :post :put :delete :head)
   :body (s/maybe {s/Any s/Any})
   :form-params  (s/maybe {s/Any s/Any})
   :query-params  (s/maybe {s/Any s/Any})
   :connection-manager (s/either PoolingClientConnectionManager PoolingHttpClientConnectionManager)}
```
in addtion the defaut opts being:
```clojure
(def default-opts
  {:as :json
   :content-type :json
   :throw-exceptions false})
```
The return schema  could be  (s/maybe {s/Any s/Any}) (I propose this schema for body, form-params and query-params for the same reason)

Reply:
For some reason this makes me nervous so I'll give it some time to sink in.